feat: :sparkles: download and import script for beetle data

K-Beicher opened the pull request:
## Description

This PR contains two scripts which will allow for data to be read into the project and then into python.

Closes #3 
Closes #10 

<!-- Please delete as appropriate: -->
This PR needs an in-depth review.

I have yet to get ruff and various other things to work correctly, so it may be that there is some formatting that isn't correct.

K-Beicher committed:
feat: add pyproject and .vscode

had to add vscode to get conventional branch to work

K-Beicher committed:
feat: add items from python project pyproject file

K-Beicher committed:
feat: create import script

K-Beicher committed:
feat:

K-Beicher committed:
Merge pull request #7 from seedcase-project/main

Import data

K-Beicher committed:
feat: added code to check for and/or create gitignore file

K-Beicher committed:
feat: update convert script with load of csv file

K-Beicher committed:
feat: add docstrings to the two scripts

K-Beicher commented:
I was certain that I chose feature for this PR, but it somehow picked build...

lwjohnst86 committed:
Merge branch 'main' into build/download-and-import-data

lwjohnst86 committed:
chore: :fire: delete MacOS hidden file

lwjohnst86 commented on download-data.py:
```diff
@@ -0,0 +1,16 @@
+import os
+import requests
+
+script_dir = os.path.dirname(os.path.abspath(__file__))
+
+folder_path = os.path.join(script_dir, '..', 'data_raw')
+
+if not os.path.exists(folder_path):
+    os.mkdir(folder_path)
+else:
+    pass
```
It's better to use pathlib for building paths, not os. Please switch it over.

lwjohnst86 commented on download-data.py:
I'm a bit confused, what is this file for and when would you use it?

K-Beicher replied:
Somehow snuck up a level, I thought I'd deleted it, but obviously not. I'll make it go away.

lwjohnst86 commented on male-seed-beetle/scripts/download-data-custom.py:
```diff
@@ -0,0 +1,65 @@
+        If the user doesn't specify either a folder or a url it will abort.
+    """
+
+# folder_name = 'male-seed-beetle' For Testing Delete Later
+
+folder_name = input("Enter the name of the main folder for the project: ")
```
In general you don't want to include interactive inputs and instead provide arguments for these inputs. Please switch this to be an argument, not an input.

lwjohnst86 commented on male-seed-beetle/scripts/download-data-custom.py:
```diff
@@ -0,0 +1,65 @@
+
+folder_name = input("Enter the name of the main folder for the project: ")
+
+script_dir = os.path.dirname(os.path.abspath(__file__))
+
+folder_path = os.path.join(script_dir, '..', '..', folder_name, 'data-raw')
+
+file_path = os.path.join(folder_path, '.gitignore')
```
This is trickier to keep track of what gets added to a file. Since we won't be doing this often (really, only once), having it in code isn't necessary.

```suggestion
```

lwjohnst86 commented on male-seed-beetle/scripts/download-data-custom.py:
```diff
@@ -0,0 +1,65 @@
+
+file_path = os.path.join(folder_path, '.gitignore')
+
+if not os.path.exists(folder_path):
+    os.mkdir(folder_path)
+else:
+    pass
+
+if not os.path.isfile(file_path):
+    with open(file_path, 'w') as file:
+        file.write('*')
+else:
+    pass
```
```suggestion
```

lwjohnst86 commented on male-seed-beetle/scripts/download-data-custom.py:
```diff
@@ -0,0 +1,65 @@
+else:
+    pass
+
+source = input("Where is the file stored, write either 'u' for url or 'f' for folder: ")
```
See comment above about this.

```suggestion
```

lwjohnst86 commented on male-seed-beetle/scripts/download-data-custom.py:
```diff
@@ -0,0 +1,65 @@
+
+source = input("Where is the file stored, write either 'u' for url or 'f' for folder: ")
+
+if source == 'u':
```
What's `u`?

lwjohnst86 commented on male-seed-beetle/scripts/download-data-custom.py:
```diff
@@ -0,0 +1,65 @@
+
+if source == 'u':
+    url = input("Write the full url here: ")
+    r = requests.get(url, allow_redirects=True)
```
It isn't good practice to use single character variables. What is `r`?

lwjohnst86 commented on male-seed-beetle/scripts/download-data-custom.py:
```diff
@@ -0,0 +1,65 @@
+        file.write(r.content)
+
+elif source == 'f':
+    folder = input("Write the full path and filename here: ") 
```
See comment above. Avoid requesting user input and instead use arguments.

lwjohnst86 commented on male-seed-beetle/scripts/download-data-custom.py:
```diff
@@ -0,0 +1,65 @@
+        with open(file_path, 'wb') as dest_file:
+            dest_file.write(source_file.read())
+
+else:
+    print("Did you type either 'u' or 'f' ")
```
When informing the user about a mistyped word or argument, use errors, not prints.

lwjohnst86 commented on male-seed-beetle/scripts/download-data.py:
See comment above, they apply to this as well.

lwjohnst86 reviewed (changes requested):
Good first start, though needs a good amount of work

K-Beicher committed:
feat: update to pathlib and remove input

K-Beicher committed:
feat: update the docstring

lwjohnst86 commented on male-seed-beetle/scripts/download-data.py:
```diff
@@ -0,0 +1,34 @@
+#testing:
+url = 'https://datadryad.org/stash/downloads/file_stream/30903'
+
+def download_data(url: str):
```
This function actually doesn't contain any code. Is that intentional?

K-Beicher replied:
I thought the requests.get and file.write at the end was code, is that something else? Would it make more sense to just move the url down and drop the def and docstring?

lwjohnst86 replied:
No, those aren't part of the function. They would need to be indented to be part of the function. I think it's fine if the script doesn't have a function in it.

K-Beicher committed:
feat: remove function from download and add to gitignore file

add a line to gitignore so that it actually syncs with github

K-Beicher committed:
feat: update docstring

lwjohnst86 committed:
Merge branch 'main' of https://github.com/seedcase-project/examples into build/download-and-import-data

lwjohnst86 committed:
chore: :wrench: set project to not be a package

lwjohnst86 committed:
chore: :heavy_plus_sign: add requests as a dependency

lwjohnst86 committed:
style: :art: reformat Python script

lwjohnst86 commented on male-seed-beetle/scripts/download-data.py:
```diff
@@ -0,0 +1,25 @@
+# testing male-seed-beetle/data-raw returned with print(f'{folder_path}')
+folder_path = resource_dir / "data-raw"
+
+url = "https://datadryad.org/stash/downloads/file_stream/30903"
```
This downloads a file that has 403 still.

K-Beicher replied:
I know, I thought I would finish it as a piece of code before just downloading this particular data set. Once we are happy with the principle of it, I can put it away so that it is ready for another project where we can download from a url.

lwjohnst86 replied:
Hmmm, it's better to have the code when we use and need the code, rather than keep code around that we aren't using. So if this doesn't work, either use another dataset from Zenodo, or include instructions in the README on where to download the file and where to store it, rather than have a script that doesn't work.

K-Beicher replied:
No problem, I'll move it to the other resource and make sure to pick a dataset which will allow us to download automatically.

K-Beicher committed:
feat: move script to new folder with data that will download

found a data set on Zenodo which will automatically download to test out the script

K-Beicher committed:
feat: update male seed beetle resource with script pointing to zenodo

it turns out that the data is also available at zenodo, so I've restored the script and updated the readme

K-Beicher committed:
feat: update .gitignore to exclude .gitignore

K-Beicher commented on seed-bank-dynamics/scripts/README.md:
```diff
@@ -0,0 +1,13 @@
+# Data set on free-living birds in Sweden
```
```suggestion
# Data set on Dung Beetle activity in rainforests
```

K-Beicher replied:
I set up the whole second folder with data from Zenodo before I realised that the data set for the seed beetles where also available from Zenodo. Hence the change of name of the second resource and the fact that I've created the second data file.

lwjohnst86 committed:
docs:

Co-authored-by: Kris Beicher <112945740+K-Beicher@users.noreply.github.com>

lwjohnst86 reviewed (approved):
Nice! Great to see the male seed beetle was on Zenodo :tada:
